jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/SaveQueryTest.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.testFramework.RunsInEdt
+import com.intellij.testFramework.runInEdtAndWait
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+import software.aws.toolkits.resources.message
+import org.assertj.core.api.Assertions.assertThat
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+
+@RunsInEdt
+class SaveQueryTest {
+    @JvmField
+    @Rule
+    val projectRule = JavaCodeInsightTestFixtureRule()
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+    private lateinit var client: CloudWatchLogsClient
+    private lateinit var view: EnterQueryName
+    private lateinit var validator: SaveQueryDialog
+
+    @Test
+    fun `Query name entered`() {
```

Comment:
The name kinda suggests that this is a happy path test. Please make this more descriptive, something like `returns an error message when a query name is not provided`. As a general rule, you'll want these to be more verbose; you don't want to have to think too hard about what the test should do if you have to fix it.

Reply:
Changed the test name